### PR TITLE
[4249] Reorganize constants

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Closes https://github.com/GetStream/stream-chat-android/issues/4249

Currently, we have all constants in the `ModelType` class. We need to split them and expose them to our clients as they can be quite handy when implementing their custom logic.

### 🛠 Implementation details

Moved `MessageType`, `AttachmentType` and `ChannelType` constants to the `client` module. Moved `MimeType` constants to internal `ui-utils` module.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/AttachmentFactory.kt [hunk 1]
```diff
@@ -45,5 +44,7 @@ public open class AttachmentFactory constructor(
         modifier: Modifier,
         attachmentState: AttachmentState,
     ) -> Unit,
-    public val textFormatter: (attachments: Attachment) -> String = Attachment::previewText,
+    public val textFormatter: (attachments: Attachment) -> String = {
+        it.title ?: it.name ?: ""
+    },
```
Comment (JcMinarro) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/AttachmentFactory.kt:49]: Why don't we use the extension?

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt [hunk 2]
```diff
@@ -144,37 +143,37 @@ class PlainTextWithFileAttachmentsMessagesComponentBrowserFragment : BaseMessage
 
         val attachmentPdf = Attachment(
             type = "file",
-            mimeType = ModelType.attach_mime_pdf,
+            mimeType = "application/pdf",
```
Comment (JcMinarro) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt:146]: Can't we use constants for those?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/ChannelType.kt [hunk 3]
```diff
@@ -14,11 +14,13 @@
...
-package io.getstream.chat.android.ui.common.extensions.internal
+package io.getstream.chat.android.client.models
 
-import com.getstream.sdk.chat.model.ModelType
-import io.getstream.chat.android.client.models.Attachment
-
-internal fun Attachment.isImage(): Boolean = type == ModelType.attach_image
-
-internal fun Attachment.isGiphy(): Boolean = type == ModelType.attach_giphy
+public object ChannelType {
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/ChannelType.kt:19]: Do we need `ChannelType` constants?

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/AttachmentFactory.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/AttachmentFactory.kt:49]: It was the last extension in the file and it was used only in one place, so I decided to move the code to the call site.

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt:146]: Hm, I moved mime type constants to the internal UI module. It it not exposed to the sample app. Any suggestions?

Comment (JcMinarro) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt:146]: Ohh, I see, this is the sample app....
If we are using it there, should it be needed by our customers as well???
What about exposing those constants as public constants? :thinking:

Comment (adasiewiczr) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/component_browser/messages/viewholder/PlainTextWithFileAttachmentsMessagesComponentBrowserFragment.kt:146]: If we decide to expose mime types we should carefully review all of them so we don't expose invalid ones like

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**